src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,140 @@
+import React, {
+  Children as ReactChildren,
+  createElement,
+  cloneElement,
+  CSSProperties,
+  Fragment,
+  useEffect,
+  useMemo,
+  useState,
+  createRef,
+  PropsWithChildren,
+} from "react";
+import { forwardRef } from "react";
+import { Children } from "~/utils/typing/children";
+import { SPACE_UNIT } from "~/utils/constants";
+import { Box } from "~/index";
+
+export type ContainerProps = PropsWithChildren<{
+  "data-test-id"?: string;
+  rows: number;
+  cols: number;
+  rowsGap?: number;
+  colsGap?: number;
+  className?: string;
+  style?: CSSProperties;
+}> & {
+  children: React.ReactElement | React.ReactElement[];
+};
+
+const Container = forwardRef<HTMLDivElement, ContainerProps>((props, ref) => {
+  const { children } = props;
```

Comment:
please follow the internal code guidelines and do not destructure `children`

Reply:
done